dashboard/client/src/pages/job/hook/useJobProgress.ts

```diff
@@ -15,21 +15,45 @@ import {
 } from "../../../type/job";
 import { TypeTaskStatus } from "../../../type/task";
 
+const enum TaskStatus {
+  PENDING_ARGS_AVAIL = "PENDING_ARGS_AVAIL",
+  PENDING_NODE_ASSIGNMENT = "PENDING_NODE_ASSIGNMENT",
+  SUBMITTED_TO_WORKER = "SUBMITTED_TO_WORKER",
+  RUNNING = "RUNNING",
+  FINISHED = "FINISHED",
+  FAILED = "FAILED",
+  UNKNOWN = "UNKNOWN",
+}
+
 const TASK_STATE_NAME_TO_PROGRESS_KEY: Record<
   TypeTaskStatus,
+  keyof TaskStatus
```

Comment:
should this just be `TaskStatus` instead of `keyof TaskStatus`?

Reply:
It works!